src/hooks/useProjectVariablesUpdate/util.ts

```diff
@@ -177,12 +177,18 @@ export const makeVariableValueOperations = ({
 
   if (newTextValues) {
     newTextValues.forEach((nV, index) => {
-      if (pendingValues[index].id !== -1) {
+      if (pendingValues[index].id !== -1 && nV.textValue) {
         operations.push({
+          existingValueId: pendingValues[index].id,
           operation: 'Update',
           valueId: pendingValues[index].id,
           value: nV,
+        });
+      } else if (pendingValues[index].id !== -1 && !nV.textValue) {
+        operations.push({
           existingValueId: pendingValues[index].id,
+          operation: 'Delete',
+          valueId: pendingValues[index].id,
```

Comment:
and adding this block. Otherwise, other changes were only to re-order object props.